src/rules/no-classfield-shadowing.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @fileoverview Disallows properties shadowed as class fields
+ * @author Michel Langeveld <https://github.com/michellangeveld>
+ */
+
+import {Rule} from 'eslint';
+import * as ESTree from 'estree';
+import {getClassFields, getPropertyMap} from '../util';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      description: 'Disallows properties shadowed as class field',
+      recommended: false,
```

Comment:
Could this be recommended? Since this helps with potential runtime errors and doesn't seem to have any drawbacks I feel like we could make this recommended.

Reply:
Yes, recommended makes sense indeed.